jwql/pull_jwql_branch.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+
+function echo_format {
+    echo "WARNING! the optional parameters should only be used during a JWQL release in production"
+    echo "Usage: $0 <branch> [-r|--reset_service] [-n|--notify <email@stsci.edu>]"
+}
+
+# Check if the required number of arguments are provided
+if [ "$#" -lt 1 ]; then
+    echo_format
+    exit 1
+fi
+
+# Set default values for optional flags
+reset=false
+notify=false
+recipient=""
+
+# Retrieve the branch_name from the command line argument
+branch_name=$1
+# Parse optional flags
+while [[ $# -gt 1 ]]; do
+    case "$2" in
+        -r|--reset_service)
+            reset=true
+            ;;
+        -n|--notify)
+            notify=true
+            recipient="$3"
+            shift
+            ;;
+        *)
+            echo "Error: Invalid option $2"
+            echo_format
+            exit 1
+            ;;
+    esac
+    shift
+done
+
+if [ "$notify" = true ]  && [ -z "$recipient" ]; then
+    echo_format
+    exit 1
+fi
+
+echo "Branch: $branch_name";
+echo "Reset: $reset";
+echo "Notify: $notify $recipient";
+
+# 1. Pull updated code from GitHub deployment branch (keep second checkout in case its already defined for some weird reason)
+git checkout -b $branch_name --track origin/$branch_name
+git checkout $branch_name
+git fetch origin $branch_name
+git pull origin $branch_name
+git fetch origin --tags
+
+# 2. Bring the server down
+if [ "$reset" = true ]; then
+    sudo /bin/systemctl stop jwql.service
+fi
+
+# 3. Install jwql
+pip install -e ..
+
+# 4. Merge Any Migrations
+python ./website/manage.py migrate
+
+# 5. Bring the server back up
+if [ "$reset" = true ]; then
```

Comment:
From what I see here, this script is also still useful on the servers as well. Locally we don't need to reset the servers nor notify anyone. Although I spamming your email might be fun @BradleySappington 😉

Reply:
lol, correct, this is still intended to be our "release procedure".  I put a warning at the top to not use the optional parameters for local use.  (I also wanted to prevent us from sending multiple emails if we ever need to run this twice in a row...not that we've ever had to do that...)

Reply:
🤫 I don't know what you're talking about here...